cmake/host-tools.cmake

```diff
@@ -27,23 +27,33 @@ else()
   endif()
 
   if(${west_version} VERSION_LESS ${MIN_WEST_VERSION})
+    if(WEST_PYTHON)
+      if(NOT (${WEST_PYTHON} STREQUAL ${PYTHON_EXECUTABLE}))
+        set(PYTHON_EXECUTABLE_OUT_OF_SYNC "Note:\n\
+  The Python version used by west is:  ${WEST_PYTHON}\n\
+  The Python version used by CMake is: ${PYTHON_EXECUTABLE}\n\
+  This might be correct, but please verify your installation.\n")
+      endif()
+    endif()
     message(FATAL_ERROR "The detected west version is unsupported.\n\
   The version was found to be ${west_version}:\n\
     ${item}\n\
   But the minimum supported version is ${MIN_WEST_VERSION}\n\
   Please upgrade with:\n\
-      pip3 install --upgrade west")
+      ${PYTHON_EXECUTABLE} -mpip install --upgrade west\n\n\
+  ${PYTHON_EXECUTABLE_OUT_OF_SYNC}")
   endif()
   # Just output information for a single version. This will still work
   # even after output is one line.
   message(STATUS "Found west: ${WEST} (found suitable version \"${west_version}\", minimum required is \"${MIN_WEST_VERSION}\")")
 
-    execute_process(
-      COMMAND ${WEST}  topdir
-      OUTPUT_VARIABLE  WEST_TOPDIR
-      OUTPUT_STRIP_TRAILING_WHITESPACE
-      WORKING_DIRECTORY ${ZEPHYR_BASE}
-      )
+  execute_process(
+    COMMAND ${PYTHON_EXECUTABLE} -c
+    "import sys; from west.app.main import main; main(['topdir'])"
```

Comment:
This is an abstraction violation as west.app is not documented API. We would essentially be requiring the `west.app.main` package to exist and have a function named `main` until the end of time or at least the EOL of this zephyr version.

I think I am OK with doing that, but only if we can update the west API page to document this, so we can treat it as a regression if it ever breaks.

Reply:
I kind of dislike it myself, so if we make `west` to be easier to import and run command, i'm fine with doing it another way. But agreed, as minimum, we should document this possibility in the API.

But what has been discovered, and also discussed on slack, is that it is unsafe to run `west` as native command with one python, and then other build script using another python.

Reply:
Sure, I am OK with doing this as long as we document the entry point, so we can maintain it. I think @pfalcon 's suggestion to make the west package runnable is another way to get around this in the future.

Reply:
Docs updated, and while looking at the docs, I decided to switch from `main([<args>])` to `WestApp().run([<args>])` as that looks more correct, imho.

Reply:
also see: https://github.com/zephyrproject-rtos/west/pull/406